core/src/banking_stage/thread_aware_account_locks.rs

```diff
@@ -310,12 +308,16 @@ impl Debug for ThreadSet {
 impl ThreadSet {
     #[inline(always)]
     pub(crate) const fn none() -> Self {
-        Self(0)
+        Self(0b0)
     }
 
     #[inline(always)]
     pub(crate) const fn any(num_threads: usize) -> Self {
-        Self(Self::as_flag(num_threads) - 1)
+        if num_threads == MAX_THREADS {
+            Self(u64::MAX)
```

Comment:
btw, i also considered `Self(!0b0)` for type agnotic code, but chose the const albeit, while being a bit off-meaning here... lol